pennylane/compiler/qjit_api.py

```diff
@@ -610,17 +611,21 @@ def _call_capture_enabled(self, *init_state):
 
         for_loop_prim = _get_for_loop_qfunc_prim()
 
-        jaxpr_body_fn = jax.make_jaxpr(self.body_fn)(0, *init_state)
+        flat_args, in_tree = jax.tree_util.tree_flatten((0, *init_state))
+        flat_fn = FlatFn(self.body_fn, in_tree)
+        jaxpr_body_fn = jax.make_jaxpr(flat_fn)(*flat_args)
 
-        return for_loop_prim.bind(
+        results = for_loop_prim.bind(
             self.lower_bound,
             self.upper_bound,
             self.step,
             *jaxpr_body_fn.consts,
-            *init_state,
+            *flat_args[1:],  # skip the initial 0
             jaxpr_body_fn=jaxpr_body_fn,
             n_consts=len(jaxpr_body_fn.consts),
         )
+        assert flat_fn.out_tree is not None
```

Comment:
What's the point of this assertion?

Reply:
Honestly, mostly to get pylance to stop complaining.

Reply:
Valid.